sklearn/setup.py

```diff
@@ -1,8 +1,36 @@
+from pathlib import Path
 import os
 
 from sklearn._build_utils import maybe_cythonize_extensions
 
 
+_DEPRECATED_MODULES = {
+    ('_mocking', 'sklearn.utils.mocking', 'sklearn.utils')
+}
+
+_DEPRECATE_TEMPLATE = """from .{module} import *  # noqa
+from ..utils.deprecated import _raise_dep_warning_if_not_pytest
+
+deprecated_path = '{deprecated_path}'
+correct_path = '{correct_path}'
+
+_raise_dep_warning_if_not_pytest(deprecated_path, correct_path)
+"""
+
+
+# Adds files that will be deprecated
+def _adds_deprecated_submodules():
```

Comment:
```suggestion
def _add_deprecated_submodules():
```

Also it would need a short description (remaning files + creating new file to keep the import)